rundeckapp/src/test/groovy/rundeck/controllers/MenuControllerSpec.groovy

```diff
@@ -1645,4 +1654,147 @@ class MenuControllerSpec extends HibernateSpec implements ControllerUnitTest<Men
+
+        UserAndRolesAuthContext test = Mock(UserAndRolesAuthContext) {
+            getUsername() >> 'test'
+            getRoles() >> new HashSet<String>(['test'])
+        }
+
+        def projectMock = Mock(IRundeckProject) {
+            getProperties() >> [:]
+            getName() >> 'aProject'
+        }
+
+        controller.executionService = Mock(ExecutionService){
+            queryQueue(_) >> [:]
+            finishQueueQuery(_,_,_) >> [:]
+        }
+
+        controller.frameworkService = Mock(FrameworkService){
+            getRundeckBase() >> ''
+            getFrameworkProject(_) >> projectMock
+            getServerUUID() >> 'uuid'
+            authorizeProjectResourceAll(_, AuthorizationUtil.resourceType('event'),['read'], 'aProject') >> false
+            1 * getAuthContextForSubject(_) >> test
+
+        }
+
+        params.project = '*'
+        request.api_version = 35
+        def action = 'read'
+
+        when:
+        def result = controller.apiExecutionsRunning()
+
+        then:
+        0 * controller.frameworkService.getAuthContextForSubjectAndProject(_, '*')
+        1 * controller.frameworkService.projectNames(_) >> ['aProject']
+        1 * controller.apiService.renderErrorFormat(_,{map->
+            map.status==401 && map.code=='api.error.execution.project.notfound'
+        })
+
+    }
+
+    def "test list all projects with an invalid and a valid project"() {
+        given:
+        controller.apiService = Mock(ApiService){
+            1 * requireApi(_,_) >> true
+        }
+
+        controller.userService = Mock(UserService){}
+
+        UserAndRolesAuthContext test = Mock(UserAndRolesAuthContext) {
+            getUsername() >> 'test'
+            getRoles() >> new HashSet<String>(['test'])
+        }
+
+        def projectMock = Mock(IRundeckProject) {
+            getProperties() >> [:]
+            getName() >> ['aProject', 'bProject']
+        }
+
+        controller.executionService = Mock(ExecutionService){
+            queryQueue(_) >> [:]
+            finishQueueQuery(_,_,_) >> [:]
+        }
+
+        controller.frameworkService = Mock(FrameworkService){
+            getRundeckBase() >> ''
+            getFrameworkProject(_) >> projectMock
+            getServerUUID() >> 'uuid'
+            authorizeProjectResourceAll(_, AuthorizationUtil.resourceType('event'),['read'], 'aProject') >> true
+            1 * getAuthContextForSubject(_) >> test
+
+        }
+
+        params.project = '*'
+        request.api_version = 35
+        def action = 'read'
+
+        when:
+        def result = controller.apiExecutionsRunning()
+
+        then:
+        0 * controller.frameworkService.getAuthContextForSubjectAndProject(_, '*')
+        1 * controller.frameworkService.projectNames(_) >> ['aProject', 'bProject']
+        0 * controller.apiService.renderErrorFormat(_,{map->
+            map.status==401 && map.code=='api.error.execution.project.notfound'
+        })
+
+    }
```

Comment:
These tests should verify that the ExecutionService.queryQueue method is called with the right projFilter value